src/client/components/DetailsModal/Details/SwedishCarDetails.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import * as Yup from 'yup'
+import { FormikProps } from 'formik'
+import { InputGroup } from 'components/inputs'
+import { TextKeyMap } from 'utils/textKeys'
+
+import { QuoteInput } from '../types'
+import { MileageInput, TextInput, ZipcodeInput } from './components/DetailInput'
+
+import { Content, ContentColumn } from './components/Details.styles'
+
+export const getSwedishCarValidationSchema = (textKeys: TextKeyMap) => {
+  return Yup.object().shape({
+    firstName: Yup.string().required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+    lastName: Yup.string().required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+    data: Yup.object({
+      street: Yup.string().required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+      zipCode: Yup.string()
+        .matches(/^[0-9]{3}[0-9]{2}$/, textKeys.GENERIC_ERROR_INPUT_FORMAT())
+        .required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+      city: Yup.string().required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+      registrationNumber: Yup.string().required(
+        textKeys.GENERIC_ERROR_INPUT_REQUIRED(),
+      ),
+      mileage: Yup.string().required(textKeys.GENERIC_ERROR_INPUT_REQUIRED()),
+      isStudent: Yup.boolean(),
+    }).required(),
+  })
+}
+
+export const SwedishCarDetails = ({
+  formikProps,
+}: {
+  formikProps: FormikProps<QuoteInput>
+}) => {
+  return (
+    <Content>
+      <ContentColumn>
+        <InputGroup>
+          <TextInput
+            name="firstName"
```

Comment:
Does it make a difference that `firstName` and `lastName` don't have `data.` like the other fields?

Reply:
Not needed since they are not nested in the data field of the passed object:

https://github.com/HedvigInsurance/web-onboarding/blob/feat/GRW-1028/car-edit-information-modal/src/client/components/DetailsModal/index.tsx#L204-L212